packages/broker/src/config/migration.ts

```diff
@@ -0,0 +1,155 @@
+import fs from 'fs'
+import { get, omitBy, set } from 'lodash'
+import { Config, getDefaultFile, getLegacyDefaultFile } from './config'
+import { isValidConfig } from './validateConfig'
+import TEST_CONFIG_SCHEMA from './config-testnet.schema.json'
+
+export const CURRENT_CONFIGURATION_VERSION = 1
+
+export const formSchemaUrl = (version: number): string => {
+    return `http://schema.streamr.com/config-v${version}.schema.json`
+}
+
+const getVersion = (config: any): number | undefined => {
+    const PATTERN = /config-v([0-9]+).schema.json$/
+    const schemaUrl = config.$schema
+    if (schemaUrl !== undefined) {
+        const match = PATTERN.exec(schemaUrl)
+        if (match !== null) {
+            return Number(match[1])
+        }
+    }
+    return undefined
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const needsMigration = (config: any): boolean => {
+    return getVersion(config) !== CURRENT_CONFIGURATION_VERSION
+}
+
+/*
+ * Migrate a Testnet3 config to the new format used in Brubeck mainnet.
+ *
+ * Some new features were added between Testnet3 and Brubeck mainnet. The migration assumes
+ * that a source config doesn't contain settings for those features. Therefore it ignores:
+ * - network.webrtcDisallowPrivateAddresses
+ * - subscriber plugin
+ */
+const convertTestnet3ToV1 = (source: any): Config => {
+    const DEFAULT_NAME = 'miner-node'
+    const target: any = {
+        $schema: formSchemaUrl(CURRENT_CONFIGURATION_VERSION)
```

Comment:
Maybe explicit set to `1`
```suggestion
        $schema: formSchemaUrl(1)
```

Reply:
Fixed